src/character_controller.h

```diff
@@ -237,15 +237,18 @@ typedef struct character_command_s
     uint16_t    sprint : 1;
 }character_command_t, *character_command_p;
 
-typedef struct character_response_s
+typedef struct character_state_s
 {
     int8_t      vertical_collide;
     int8_t      horizontal_collide;
-    uint16_t    kill : 1;
-    uint16_t    burn : 1;
     uint16_t    slide : 2;
     uint16_t    step_z : 2;
-}character_response_t, *character_response_p;
+    uint16_t    kill : 1;
```

Comment:
Most of these are used as booleans, which is not clear from the code; others are used as enums, and others are used as bitfields. If it's not possible (or too much work) to use proper data types, please add at least a few comments here what each member is supposed to contain and which effects it should have.